### PR TITLE
refactor: bubble up snapshot error

### PR DESCRIPTION
**Pull request by y-lakhdar**
## Proposed changes

Bubble up any error happening while performing snapshot operations to handle it at the command level

This was necessary to inform the user about the next steps if the snapshot operation takes too long.
image

## Testing

- [ ] Unit Tests:
  
- [ ] Functionnal Tests:
  
- [ ] Manual Tests:
  

-----
CDX-463

### CODE REVIEW

File: packages/cli/src/commands/org/config/pull.ts [hunk 1]
```diff
@@ -31,15 +39,49 @@ export default class Pull extends Command {
       multiple: true,
       description: 'The resources types to pull from the organization.',
     }) as IOptionFlag<(keyof typeof ResourceType)[]>,
+    snapshotId: flags.string({
+      char: 's',
+      exclusive: ['resourceTypes'],
+      description:
+        'The unique identifier of the target snapshot to pull. If not specified, a new snapshot wil be created',
```
Comment (jpmarceau) [inline packages/cli/src/commands/org/config/pull.ts:46]:
```suggestion
        'The unique identifier of the snapshot to pull. If not specified, a new snapshot will be created.',
```

File: packages/cli/src/commands/org/config/pull.ts [hunk 2]
```diff
@@ -31,15 +39,49 @@ export default class Pull extends Command {
...
+    this.displayAdditionalErrorMessage(err);
+  }
+
+  private displayAdditionalErrorMessage(err?: Error) {
+    if (err instanceof SnapshotOperationTimeoutError) {
+      const snapshot = err.snapshot;
+      cli.info(
+        dedent`
+
+          Once the snapshot is created, you can pull it with the following command
```
Comment (jpmarceau) [inline packages/cli/src/commands/org/config/pull.ts:80]:
```suggestion
          Once the snapshot is created, you can pull it with the following command:
```

File: packages/cli/src/lib/snapshot/snapshotCommon.ts [hunk 3]
```diff
@@ -76,6 +77,23 @@ export async function getTargetOrg(config: Config, target?: string) {
...
+  } else {
+    throw err;
+  }
+}
+
+function operationGettingTooMuchTimeMessage(snapshot: Snapshot): string {
+  return dedent`Snapshot ${
+    snapshot.latestReport.type
+  } operation is taking a long time to complete.
+  Run the following command to monitor the operation
```
Comment (jpmarceau) [inline packages/cli/src/lib/snapshot/snapshotCommon.ts:92]:
```suggestion
  Run the following command to monitor the operation:
```

**Review by jpmarceau — APPROVED** [review]

File: packages/cli/src/commands/org/config/pull.ts [hunk 4]
```diff
@@ -31,15 +39,49 @@ export default class Pull extends Command {
...
+
+  private displayAdditionalErrorMessage(err?: Error) {
+    if (err instanceof SnapshotOperationTimeoutError) {
+      const snapshot = err.snapshot;
+      cli.info(
+        dedent`
+
+          Once the snapshot is created, you can pull it with the following command:
+
+              ${blueBright`coveo org:config:pull ${snapshot.id}`}`
```
Comment (louis-bompart) [inline packages/cli/src/commands/org/config/pull.ts:82]: wow thats lot of spacing innit?

File: packages/cli/src/lib/snapshot/snapshotCommon.ts [hunk 5]
```diff
@@ -76,6 +77,23 @@ export async function getTargetOrg(config: Config, target?: string) {
   return cfg.organization;
 }
 
+export function handleSnapshotError(err?: Error) {
```
Comment (louis-bompart) [inline packages/cli/src/lib/snapshot/snapshotCommon.ts:80]: lgtm, just add some unit test for this utils ;)

**Review by olamothe — APPROVED** [review]

File: packages/cli/src/commands/org/config/pull.ts (hunk 4 above)
Comment (y-lakhdar) [reply packages/cli/src/commands/org/config/pull.ts:82]: I can reduce it to 2 spaces

Comment (louis-bompart) [reply packages/cli/src/commands/org/config/pull.ts:82]: > I can reduce it to 2 spaces

can you show some screenshot for current and with just 2 space (I'm talking about the breaklines btw more than the spacing itself, sorry was unclear)

File: packages/cli/src/lib/snapshot/snapshotCommon.ts (hunk 5 above)
Comment (y-lakhdar) [reply packages/cli/src/lib/snapshot/snapshotCommon.ts:80]: It's kinda tricky to test this use case in the UT. I planned to cover that use case in the E2E tests

File: packages/cli/src/commands/org/config/pull.ts (hunk 4 above)
Comment (y-lakhdar) [reply packages/cli/src/commands/org/config/pull.ts:82]: image

VS 

image

**Pull request merged**